### PR TITLE
fix: default repo and tag

### PR DESCRIPTION
**Pull request by pmengelbert**
- Update version.sh script to set default tag with ldflags
- Use default tag in default config
- Make image repos more typo-resistant by using a default repo, settable by ldflags

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: version/version.go [hunk 1]
```diff
@@ -14,6 +14,14 @@ var (
 	vcsCommit string
 )
 
+var (
+
+	// DefaultTag is the default repo for images
+	DefaultRepo = "ghcr.io/azure"
+	// DefaultTag is the default tag for images
+	DefaultTag = "latest"
```
Comment (sozercan) [inline version/version.go]: should we use `buildVersion` https://github.com/Azure/eraser/blob/main/version/version.go#L10?

Comment (pmengelbert) [reply version/version.go]: `buildversion` has the commit hash attached, example: `-X 'github.com/Azure/eraser/version.buildVersion=v1.0.0-beta.3.24-049b7d2c902a3d'`

Comment (sozercan) [reply version/version.go]: if you are on a git tag, it shouldn't show the commit. 

for example:
```shell
$ git checkout v1.0.0-rc.1
...
$ ./build/version.sh  
-X 'github.com/Azure/eraser/version.buildTime=1675117790' -X 'github.com/Azure/eraser/version.vcsCommit=34736043ef5cef29774745cd8e95029dde48045d' -X 'github.com/Azure/eraser/version.vcsState=clean' -X 'github.com/Azure/eraser/version.buildVersion=v1.0.0-rc.1' -X 'github.com/Azure/eraser/version.vcsMajor=1' -X 'github.com/Azure/eraser/version.vcsMinor=0'
```

Comment (pmengelbert) [reply version/version.go]: But then during development, deploying the helm chart in `manifest_staging/charts/eraser` will default to using `v.1.0.0-beta.2.24-049b7d2c902a3d` while `make docker-build-eraser` will produce images with the tag `v1.0.0-beta.3`.

Then the developer needs to manually match up the tags every time they deploy the helm chart to test it, which is kinda surprising behavior.

Another thing we could do is provide a default tags in `values.yaml` that match VERSION in the Makefile, but we'd need to update it every time there's a new release. We could just change the release script to update them when a new version releases.

File: build/version.sh [hunk 2]
```diff
@@ -88,6 +88,8 @@ version::ldflags() {
         add_ldflag "buildVersion" "$1"
     fi
 
+    add_ldflag "DefaultTag" "$1"
```
Comment (sozercan) [inline build/version.sh]: is this used?

File: manifest_staging/charts/eraser/values.yaml [hunk 3]
```diff
@@ -32,9 +32,9 @@ runtimeConfig:
   components:
     collector:
       enabled: true
-      image: {}
+      image:
         # repo: ""
-        # tag: ""
+        tag: "v1.0.0-beta.3"
```
Comment (sozercan) [inline manifest_staging/charts/eraser/values.yaml:37]: i think we'll need to change these to rc 1 when we cherry pick

**Review by sozercan — APPROVED** [review]

File: build/version.sh (hunk 2 above)
Comment (pmengelbert) [reply build/version.sh]: Not anymore, I'll remove

**Pull request merged**